### PR TITLE
AP-2406 application detail report

### PR DESCRIPTION
**Pull request by stephenrichards**
## Add new columns to the applications detail report for MP

[Link to story](https://dsdmoj.atlassian.net/browse/AP-2406)

- Added new columns
- Converted tests requiring multiple proceedings to use the new factory traits
- Added scope to ApplicationProceedingType to order records in created_at sequence
- Added rake task `mp:pop` to turn on multiple proceedings and populate section 8 proceeding types

## Checklist

Before you ask people to review this PR:

- [ ] Tests and rubocop should be passing: `bundle exec rake`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [ ] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [ ] The PR description should say what you changed and why, with a link to the JIRA story.
- [ ] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/models/legal_aid_application.rb [hunk 1]
```diff
@@ -152,16 +149,17 @@ def find_or_create_lead_proceeding_type
   end
 
   def application_proceedings_by_name
-    types = application_proceeding_types.map do |application_proceeding_type|
+    # The name of this method is misleading - as they are no longer sorted by name, but by the order in
```
Comment (colinbruce) [inline app/models/legal_aid_application.rb:152]: Should we have a ticket to refactor this?

**Review by colinbruce — APPROVED** [review]
LGTM :+1:

One possible refactoring observation

File: app/models/legal_aid_application.rb (hunk 1 above)
Comment (stephenrichards) [reply app/models/legal_aid_application.rb:152]: I'll do the refactor now.

Comment (stephenrichards) [reply app/models/legal_aid_application.rb:152]: I've just changed the comment  to:
```
    # returns an array of OpenStructs containing:
    # - name of the proceeding type
    # - meaning of the proceeding type
    # - the ApplicationProceedingType
    #
    # in the order they were added to the LegalAidApplication
    # 
```
Probably doesn't need renaming after all.

Comment (colinbruce) [reply app/models/legal_aid_application.rb:152]: Cool 👍

**Pull request merged**